tests/unit/test.program.js

```diff
@@ -898,17 +898,19 @@ describe('program.defaultVersionGetter', () => {
   });
 
   it('returns git commit information in development', function () {
-    return fs.exists(path.join(projectRoot, '.git')).then(async (exists) => {
-      if (!exists) {
+    return fs.access(path.join(projectRoot, '.git')).then(
+      async () => {
+        const commit = `${git.branch(projectRoot)}-${git.long(projectRoot)}`;
+        const testBuildEnv = { globalEnv: 'development' };
+        assert.equal(
+          await defaultVersionGetter(projectRoot, testBuildEnv),
+          commit,
+        );
+      },
+      () => {
         this.skip();
-      }
-      const commit = `${git.branch(projectRoot)}-${git.long(projectRoot)}`;
-      const testBuildEnv = { globalEnv: 'development' };
-      assert.equal(
-        await defaultVersionGetter(projectRoot, testBuildEnv),
-        commit,
-      );
-    });
+      },
+    );
   });
```

Comment:
Nit: not really mandatory, but given that we need to touch this test case a bit then we may as well rewrite into an async function, and make it slightly more readable:

```js
  it('returns git commit information in development', async function () {
    try {
      await fs.access(path.join(projectRoot, '.git'));
    } catch {
      this.skip();
      return;
    }
    const commit = `${git.branch(projectRoot)}-${git.long(projectRoot)}`;
    const testBuildEnv = { globalEnv: 'development' };
    assert.equal(
      await defaultVersionGetter(projectRoot, testBuildEnv),
      commit,
    );
  });
```

but I'd also like to merge this sooner rather than later, and in practice the test would be basically the same, and so we can eventually do that later after this is merged.

Reply:
Definitely, in my PRs here I just limited my changes to the bare minimum to avoid unnecessary bikeshedding. 

I think both this repo and addons-linter would benefit from a strict lint config and a whole bunch of rules (like `eslint-plugin-unicorn`) with autofixes. 

Your suggestion is https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/prefer-await-to-then.md